table/block_based/block_based_table_reader.cc

```diff
@@ -3178,19 +3178,44 @@ Status BlockBasedTable::GetKVPairsFromDataBlocks(
   return Status::OK();
 }
 
+// It's an output stream helper to build string to WritableFile.
+// Which auto flushes to file when the object is destroyed (like RAII), the
+// flush error is ignored, unless manually call `flush()`.
+class WritableFileStringStreamHelper {
+ public:
+  WritableFileStringStreamHelper(WritableFile* writable_file)
+      : file_(writable_file) {}
+
+  ~WritableFileStringStreamHelper() { assert(flush()); }
+
+  bool flush() {
+    Status s = file_->Append(stream_buff_.str());
+    stream_buff_.clear();
+    return s.ok();
+  }
+
+  // Simple template just forward the input to stringstream buffer
+  template <typename T>
+  WritableFileStringStreamHelper& operator<<(const T& t) {
```

Comment:
I don't see any manual or automatic flushes. So doesn't this build an in-memory textual dump of the whole file before writing it out? That's not practical/safe for very large SST files.

Reply:
Good point. I missed that. Added a flush threshold and a simple test for that.

Reply:
Good point. I removed the buffering. Now we have an adapter class to make `WritableFile` working with `ostream`.